UnityProject/Assets/Scripts/Shuttles/EscapeShuttleConsole.cs

```diff
@@ -72,17 +83,75 @@ private void FinishHack(HandApply interaction)
 
 			beenEmagged = true;
 
-			if (GameManager.Instance.ShuttleSent) return;
+			DepartShuttle();
+		}
 
-			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch Triggered!</b></size></color>\n\n",
-				MatrixManager.MainStationMatrix);
+		private void RegisterEarlyShuttleLaunch(IDCard card, PlayerScript performer)
+		{
+			if (GameManager.Instance.ShuttleSent)
+			{
+				Chat.AddExamineMsg(performer.gameObject, "The shuttle is already moving!");
+				return;
+			}
+			if (registeredIDs.Contains(card))
+			{
+				Chat.AddExamineMsg(performer.gameObject, "You've already done this!");
+				return;
+			}
+			registeredIDs.Add(card);
 
-			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch Triggered!</b></size></color>\n\n",
+			if (IsHighAlert() && registeredIDs.Count >= 2)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			if (registeredIDs.Count >= 4)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			AnnounceRemainingSwipesRequired();
+		}
+
+		private bool IsHighAlert()
+		{
+			return GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red
+				or CentComm.AlertLevel.Delta;
+		}
```

Comment:
I meant something like this:
```suggestion
		private int RequiredSwipesEarlyLaunch => GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta ? 2 : 4;
```
Then you can just compare against that in both cases, and you don't configure the `2` and `4` in two places.